tutorials/llm/slimpajama/data_pipeline.ipynb

```diff
@@ -0,0 +1,328 @@
+    "    output_dir=\"/data/slimpajama_megatron\",\n",
+    "    tokenizer_model=\"/data/tokenizer/tokenizer.model\",\n",
+    "    tokenizer_library=\"sentencepiece\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preprocess_task"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Putting it all together\n",
+    "\n",
+    "Now that all the tasks are configured, lets define an executor to run them on and an experiment to run them sequeuntially. \n",
+    "\n",
+    "> [!NOTE]\n",
+    "> Each task can be run individually, or in any combination. The notebook runs all tasks sequentially. To remove a task, just remove the corresponding `exp.add(...)` for that corresponding task.\n",
+    "> This customization is handy if you already have `jsonl` files processed, for example, from NeMo-Curator."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's define a local executor to run the experiment locally.\n",
+    "def docker_executor(host_data_path: str):\n",
+    "    packager = run.GitArchivePackager(subpath=\"examples/llm/slimpajama\") # This will package all code inside the folder. NOTE: only committed changes are packaged, so if you make a change, make sure to commit it.\n",
+    "    executor = run.DockerExecutor(\n",
+    "        packager=packager,\n",
+    "        ipc_mode=\"host\",\n",
+    "        shm_size=\"30g\",\n",
+    "        env_vars={\"PYTHONUNBUFFERED\": \"1\"},\n",
+    "        volumes=[f\"{host_data_path}:/data\"],\n",
+    "        container_image=\"python:3.11\",\n",
+    "        ulimits=[\"memlock:-1\", \"stack:67108864\"],\n",
+    "    )\n",
+    "    return executor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "executor = docker_executor(host_data_path=\"/home/hemild/dev/data\")\n",
+    "with run.Experiment(\"slimpajama-data-pipeline\") as exp:\n",
+    "    exp.add(download_task, name=\"download_slimpajama\", executor=executor)\n",
+    "\n",
+    "    # Use NeMo image for the remaining tasks\n",
+    "    executor.container_image = \"nvcr.io/nvidia/nemo:dev\"\n",
+    "    exp.add(extract_task, name=\"extract_slimpajama\", executor=executor)\n",
+    "\n",
+    "    # examples/llm/slimpajama is automatically mounted to /nemo_run/code\n",
+    "    exp.add(concat_task, name=\"concat_slimpajama\", executor=executor)\n",
+    "    exp.add(preprocess_task, name=\"preprocess_slimpajama\", executor=executor)\n",
+    "\n",
+    "    exp.run(sequential=True, tail_logs=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the experiment runs succesfully, you will see the bin and idx files as shown below. These can directly be used in NeMo and Megatron Data Loaders."
```

Comment:
fix spelling, fix capitallization, and add "files" to sentence

If the experiment runs successfully, you will see the BIN and IDX files as shown below. These files can directly be used in NeMo and Megatron Data Loaders."